tests/components/slider-test.js

```diff
@@ -0,0 +1,154 @@
+jest
+.dontMock('../../js/components/slider')
+.dontMock('../../js/components/utils')
+.dontMock('../../js/constants/constants')
+.dontMock('classnames');
+
+var React = require('react');
+var ReactDOM = require('react-dom');
+var TestUtils = require('react-addons-test-utils');
+var sinon = require('sinon');
+var Utils = require('../../js/components/utils');
+var Slider = require('../../js/components/slider');
+var CONSTANTS = require('../../js/constants/constants');
+
+describe('Slider', function() {
+  var props, node, tree, component, element;
+
+  // Using ReactDOM.render instead of test utils in order to allow re-render to simulate props update
+  function renderComponent() {
+    tree = ReactDOM.render(<Slider {...props} />, node);
+    component = TestUtils.findRenderedComponentWithType(tree, Slider);
+    element = ReactDOM.findDOMNode(component);
+  }
+
+  beforeEach(function() {
+    node = document.createElement('div');
+    tree = null;
+    component = null;
+    element = null;
+    props = {
+      value: 0,
+      minValue: 0,
+      maxValue: 1.0,
+      step: 0.1,
+      onChange: function(value) {
+        props.value = value;
+      },
+      ariaLabel: 'ariaLabel',
+      usePercentageForAria: true,
+      settingName: 'setting',
+      className: 'class',
+      itemRef: 'ref'
+    };
+  });
+
+  it('should render a Slider', function() {
+    renderComponent();
+    expect(component).toBeTruthy();
+  });
+
+  it('should render ARIA attributes using actual values when usePercentageForAria = false', function() {
+    props.value = 0.5;
+    props.usePercentageForAria = false;
+    renderComponent();
+    expect(element.getAttribute('aria-label')).toBe(props.ariaLabel);
+    expect(element.getAttribute('aria-valuemin')).toBe('0');
+    expect(element.getAttribute('aria-valuemax')).toBe('1');
+    expect(element.getAttribute('aria-valuenow')).toBe('0.5');
+    expect(element.getAttribute('aria-valuetext')).toBe('0.5 ' + props.settingName);
+    expect(element.getAttribute('tabindex')).toBe('0');
+    expect(element.getAttribute('role')).toBe(CONSTANTS.ARIA_ROLES.SLIDER);
+  });
+
+  it('should render ARIA attributes using percentage mode when usePercentageForAria = true', function() {
+    props.value = 0.5;
+    props.usePercentageForAria = true;
+    renderComponent();
+    expect(element.getAttribute('aria-label')).toBe(props.ariaLabel);
+    expect(element.getAttribute('aria-valuemin')).toBe('0');
+    expect(element.getAttribute('aria-valuemax')).toBe('100');
+    expect(element.getAttribute('aria-valuenow')).toBe('50');
+    expect(element.getAttribute('aria-valuetext')).toBe('50% ' + props.settingName);
+  });
+
+  it('should set a default focus id if none is provided', function() {
+    renderComponent();
+    expect(element.getAttribute(CONSTANTS.KEYBD_FOCUS_ID_ATTR).length).toBe(10);
```

Comment:
Where does the number 10 come from?

Reply:
This is just the length of the generated strings. Since they're random I can't check the actual value, but this is a bit better than checking that it's truthy.

Reply:
Sounds good